src/main/java/com/intuit/graphql/orchestrator/batch/GraphQLServiceBatchLoader.java

```diff
@@ -117,15 +118,17 @@ private CompletableFuture<List<DataFetcherResult<Object>>> load(List<DataFetchin
 
     Map<String, FragmentDefinition> mergedFragmentDefinitions = new HashMap<>();
 
+    //Possibly add a complex object to hold these two
     MultiValuedMap<String, GraphqlErrorException> queryRedactErrorsByKey = new ArrayListValuedHashMap<>();
+    Set<String> removedFragments = new HashSet<>();
```

Comment:
might have been covered but just checking.  If a fragment is referred to by another fragment, those fragments should be updated as well.

Reply:
That case is already taken care of